core/drivers/zynqmp_csu_puf.c

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2021 Foundries.io Ltd.
+ *
+ * Jorge Ramirez-Ortiz <jorge@foundries.io>
+ */
+#include <drivers/zynqmp_csu.h>
+#include <drivers/zynqmp_csu_aes.h>
+#include <drivers/zynqmp_csu_puf.h>
+#include <initcall.h>
+#include <io.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+
+#define PUF_CMD_OFFSET			0x00
+#define PUF_CFG0_OFFSET			0x04
+#define PUF_CFG1_OFFSET			0x08
+#define PUF_SHUT_OFFSET			0x0C
+#define PUF_STATUS_OFFSET		0x10
+#define PUF_WORD_OFFSET			0x18
+
+#define PUF_REGISTRATION		1
+#define PUF_REGENERATION		4
```

Comment:
This value is a bit trickier... the Xilinx code does not recognize the value, register map does recognize it -- I assume this is how you have been advised to use?

https://github.com/Xilinx/embeddedsw/blob/xlnx_rel_v2021.1/lib/sw_services/xilpuf/src/xpuf_hw.h#L60-L61

The used values are not then publicly documented...

Reply:
https://github.com/Xilinx/embeddedsw/blob/xlnx_rel_v2021.1/lib/sw_services/xilskey/src/include/xilskey_eps_zynqmp_puf.h#L63